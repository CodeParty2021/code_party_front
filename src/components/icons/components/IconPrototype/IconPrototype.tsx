import { Properties } from "csstype";
import React, { ComponentProps } from "react";
import { ReactSVG } from "react-svg";
import styled from "styled-components";

export const IconsDir = () => "icons";

type IconStyleProps = {
  display?: "inline-block" | "block";
  size?: ComponentProps<typeof ReactSVG>["width"];
  fill?: Properties["color"];
};

const IconStyle = styled(ReactSVG)<IconStyleProps>`
  display: ${(props) => props.display};
  width: ${(props) => props.size};
  height: ${(props) => props.size};
  fill: ${(props) => props.fill};
  & > span,
  & > div,
  & > svg {
    display: block;
    width: 100%;
    height: 100%;
  }
`;

IconStyle.defaultProps = {
  display: "inline-block",
  size: "16px",
};

type Props = {
  filename: ComponentProps<typeof ReactSVG>["src"];
  wrapper?: ComponentProps<typeof ReactSVG>["wrapper"];
} & IconStyleProps;

export const IconPrototype: React.FC<Props> = ({
  filename,
  wrapper = "span",
  ...props
}) => {
  return (
    <IconStyle
      src={filename}
      wrapper={wrapper}
      beforeInjection={(svg: any) => {
        svg.setAttribute(
          "style",
          `display: block; width: ${props.size}; height: ${props.size};`
        );
      }}
      {...props}
    />
  );
};
