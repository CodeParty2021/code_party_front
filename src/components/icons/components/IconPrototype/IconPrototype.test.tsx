import React from "react";
import { shallow } from "enzyme";
import { IconPrototype, IconsDir } from "./IconPrototype";

describe("<IconPrototype />", () => {
  it("snapshot test", () => {
    const wrapper = shallow(
      <IconPrototype
        filename={`${IconsDir()}/adjust_16.svg`}
        size={16}
        fill="red"
        display="block"
        wrapper="span"
      />
    );

    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
