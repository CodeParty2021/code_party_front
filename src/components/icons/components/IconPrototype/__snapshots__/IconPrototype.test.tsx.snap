// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconPrototype /> snapshot test 1`] = `
Array [
  <Styled(ReactSVG)
    beforeInjection={[Function]}
    display="block"
    size={16}
    src="icons/adjust_16.svg"
    wrapper="span"
  />,
]
`;
